x/mongo/driver/description/server.go

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) MongoDB, Inc. 2024-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package description
+
+import (
+	"fmt"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/address"
+	"go.mongodb.org/mongo-driver/tag"
+)
+
+// ServerKind represents the type of a single server in a topology.
+type ServerKind uint32
```

Comment:
Is there any advantage to `ServerKind` being a `uint32` instead of a `string`?

See the similar to the comment about `TopologyKind` for more details.

Reply:
I've created [GODRIVER-3234](https://jira.mongodb.org/browse/GODRIVER-3234) for this work.